src/lroImpl.ts

```diff
@@ -1,136 +1,18 @@
-
-/**
- * We need to selectively deserialize our responses, only deserializing if we
- * are in a final Lro response, not deserializing any polling non-terminal responses
- */
-export function shouldDeserializeLro(lroResourceLocationConfig?: string) {
-  let initialOperationInfo: LroResponseInfo | undefined;
-  let isInitialRequest = true;
-
-  return (response: any) => {
-    if (response.status < 200 || response.status >= 300) {
-      return true;
-    }
-
-    if (!initialOperationInfo) {
-      initialOperationInfo = getLroData(response);
-    } else {
-      isInitialRequest = false;
-    }
-
-    if (
-      initialOperationInfo.azureAsyncOperation ||
-      initialOperationInfo.operationLocation
-    ) {
-      return (
-        !isInitialRequest &&
-        isAsyncOperationFinalResponse(
-          response,
-          initialOperationInfo,
-          lroResourceLocationConfig
-        )
-      );
-    }
-
-    if (initialOperationInfo.location) {
-      return isLocationFinalResponse(response);
-    }
-
-    if (initialOperationInfo.requestMethod === "PUT") {
-      return isBodyPollingFinalResponse(response);
-    }
-
-    return true;
-  };
-}
-
-function isAsyncOperationFinalResponse(
-  response: any,
-  initialOperationInfo: LroResponseInfo,
-  lroResourceLocationConfig?: string
-): boolean {
-  const status: string = response.parsedBody?.status || "Succeeded";
-  if (!terminalStates.includes(status.toLowerCase())) {
-    return false;
-  }
-
-  if (initialOperationInfo.requestMethod === "DELETE") {
-    return true;
-  }
-
-  if (
-    initialOperationInfo.requestMethod === "PUT" &&
-    lroResourceLocationConfig &&
-    lroResourceLocationConfig.toLowerCase() === "azure-asyncoperation"
-  ) {
-    return true;
-  }
-
-  if (
-    initialOperationInfo.requestMethod !== "PUT" &&
-    !initialOperationInfo.location
-  ) {
-    return true;
-  }
-
-  return false;
-}
-
-function isLocationFinalResponse(response: any): boolean {
-  return response.status !== 202;
-}
-
-function isBodyPollingFinalResponse(response: any): boolean {
-  const provisioningState: string =
-    response.parsedBody?.properties?.provisioningState || "Succeeded";
-
-  if (terminalStates.includes(provisioningState.toLowerCase())) {
-    return true;
-  }
-
-  return false;
-}
-
-interface LroResponseInfo {
-  requestMethod: string;
-  azureAsyncOperation?: string;
-  operationLocation?: string;
-  location?: string;
-}
-
-function getLroData(result: any): LroResponseInfo {
-  return {
-    azureAsyncOperation: result.headers.get("azure-asyncoperation"),
-    operationLocation: result.headers.get("operation-location"),
-    location: result.headers.get("location"),
-    requestMethod: result.request.method
-  };
-}
-
 export class LroImpl<T> implements LongRunningOperation<T> {
   constructor(
-    private sendOperationFn: SendOperationFn<T>,
+    private sendOperationFn: (args: any, spec: any) => Promise<LroResponse<T>>,
```

Comment:
Is there a more specific type we can use instead of any?

Reply:
Good point, I am addressing this in https://github.com/Azure/autorest.typescript/pull/1118.